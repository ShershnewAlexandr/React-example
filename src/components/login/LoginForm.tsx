import React from 'react';
import { Link } from 'react-router-dom';
import { Field } from 'redux-form/immutable';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import TextInput from '../inputs/TextInput';
import Loader from '../loaders/Loader';
import { LoginFormContainerProps } from '../../containers/login/LoginFormContainer';
import { routes } from '../../utils/constants';
import './LoginFormStyles.css';

function LoginForm(props: LoginFormContainerProps) {
  const { handleSubmit, submitting } = props;
  const { t } = useTranslation();

  return (
    <form onSubmit={handleSubmit}>
      <Grid
        container
        direction="column"
        justify="center"
        spacing={2}
        alignItems="center"
      >
        <Grid item container justify="flex-start">
          <Grid item>
            <span className="signin-text">{t('login.signIn')}</span>
          </Grid>
        </Grid>
        <Grid container item spacing={4}>
          <Grid item xs={6}>
            <button type="button" className="social-btn">
              <Grid
                container
                direction="row"
                justify="space-around"
                alignItems="center"
              >
                <Grid item>
                  <img
                    className="telega-img"
                    src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBMaYOVMQ7Vi3nJW9CDkeoEgxi1Cm51qakpO13zwNnPi62NYJI"
                    alt="telega"
                  />
                </Grid>
                <Grid item>{t('login.telegram')}</Grid>
              </Grid>
            </button>
          </Grid>
          <Grid item xs={6}>
            <button type="button" className="social-btn">
              <Grid
                container
                direction="row"
                justify="space-around"
                alignItems="center"
              >
                <Grid item>
                  <img
                    className="vk-img"
                    src="https://cdn.icon-icons.com/icons2/1121/PNG/512/1486147202-social-media-circled-network10_79475.png"
                    alt="vk"
                  />
                </Grid>
                <Grid item>{t('login.vk')}</Grid>
              </Grid>
            </button>
          </Grid>
        </Grid>
        <Grid container item>
          <Field
            name="login"
            component={TextInput}
            type="text"
            placeholder={t('login.username')}
            disabled={submitting}
          />
        </Grid>
        <Grid container item>
          <Field
            name="password"
            component={TextInput}
            type="password"
            placeholder={t('login.password')}
            isPassword
            disabled={submitting}
          />
        </Grid>
        <Grid
          container
          item
          xs={12}
          direction="row"
          justify="space-between"
          alignItems="center"
        >
          <Grid container item xs={5} alignItems="center">
            <Grid item>
              <Field
                name="rememberMe"
                component="input"
                type="checkbox"
                className="login-checkbox"
                disabled={submitting}
              />
            </Grid>
            <Grid item>
              <span>{t('login.rememberMe')}</span>
            </Grid>
          </Grid>
          <Grid item xs={4}>
            {t('login.forgotPassword')}
          </Grid>
        </Grid>
        <Grid item xs={12}>
          <button
            type="submit"
            className={submitting ? 'button disabled' : 'button'}
          >
            {t('login.login')}
          </button>
        </Grid>
        <Grid item xs={12}>
          <Link to={routes.REGISTER}>
            <span className="no-accaunt">{t('login.noAccount')}</span>
          </Link>
          <span>{t('login.signIn')}</span>
        </Grid>
        {submitting && <Loader />}
      </Grid>
    </form>
  );
}

export default LoginForm;
