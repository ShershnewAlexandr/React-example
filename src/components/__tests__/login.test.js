import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import LoginForm from '../login/LoginForm';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('login', () => {
  const props = {
    handleSubmit: jest.fn(),
  };

  const loginForm = shallow(<LoginForm {...props} />);

  it('snapshot', () => {
    expect(toJson(loginForm)).toMatchSnapshot();
  });

  it('should catch submit', () => {
    loginForm.find('form').simulate('submit');
    expect(props.handleSubmit.mock.calls.length).toBe(1);
  });
});
