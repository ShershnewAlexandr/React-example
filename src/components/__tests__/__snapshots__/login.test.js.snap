// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    direction="column"
    justify="center"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <span
          className="signin-text"
        >
          login.signIn
        </span>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <button
          className="social-btn"
          type="button"
        >
          <WithStyles(ForwardRef(Grid))
            alignItems="center"
            container={true}
            direction="row"
            justify="space-around"
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <img
                alt="telega"
                className="telega-img"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBMaYOVMQ7Vi3nJW9CDkeoEgxi1Cm51qakpO13zwNnPi62NYJI"
              />
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              login.telegram
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </button>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={6}
      >
        <button
          className="social-btn"
          type="button"
        >
          <WithStyles(ForwardRef(Grid))
            alignItems="center"
            container={true}
            direction="row"
            justify="space-around"
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <img
                alt="vk"
                className="vk-img"
                src="https://cdn.icon-icons.com/icons2/1121/PNG/512/1486147202-social-media-circled-network10_79475.png"
              />
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              login.vk
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </button>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
    >
      <Field
        component={[Function]}
        name="login"
        placeholder="login.username"
        type="text"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
    >
      <Field
        component={[Function]}
        isPassword={true}
        name="password"
        placeholder="login.password"
        type="password"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="row"
      item={true}
      justify="space-between"
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        item={true}
        xs={5}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Field
            className="login-checkbox"
            component="input"
            name="rememberMe"
            type="checkbox"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <span>
            login.rememberMe
          </span>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={4}
      >
        login.forgotPassword
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <button
        className="button"
        type="submit"
      >
        login.login
      </button>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Link
        to="/register"
      >
        <span
          className="no-accaunt"
        >
          login.noAccount
        </span>
      </Link>
      <span>
        login.signIn
      </span>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</form>
`;
